### ISSUE TITLE
Home App under iOS 11.2.1 does not find device

### ISSUE THREAD

**Issue by drtp** [marcoraddatz/homebridge-docker#42]
Hello,

I am currently using the Homebridge on a Raspberry Pi 3 in combination with the following plugins:

homebridge-config-ui
homebridge-homematic
homebridge-broadlink-rm
homebridge-http-simple-switch
homebridge-fritz

Everything is up to date and works fine so far. 

Now, I planned to move the Homebridge into a Docker container on my DS718+. The installation worked flawlessly and the Homebridge started without any error message (except from the well known warnings). See [here](http://forum.iobroker.net/viewtopic.php?f=17&t=6921#p109661) and [here](https://forum.smartapfel.de/forum/thread/979-homebridge-im-docker-auf-synololgy-diskstation-home-app-unter-ios-11-2-1-findet/) for more details. However, my problem is that in contrast to the Raspi installation now the Home App under iOS 11.2.1 can not find the respective device anymore. I have installed no firewall under DSM 6.1. Do you have any idea what to do?

Cheers,

Thorsten

**Comment by drtp** [comment]
Hi again,

I just read something about problems with the Synology Bonjour Service [here](https://github.com/marcoraddatz/homebridge-docker/issues/7). The solution seems to be a modified .env file with respect to DS_HOSTNAME. I will try with this one and let you know, if it works:

```
# Options are all valid timezones.
# Get a list from here: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
# Default timezone is "Europe/Berlin"
HOMEBRIDGE_TIMEZONE=Europe/Berlin

# The value of the DS_HOSTNAME environment should exactly match the server name as shown under
# "Synology DSM Control Panel" -> "Info Centre" -> "Server name", it should contain no spaces or special characters.
# Default is unset.
# DO ONLY SET IF YOU GOT PROBLEMS RUNNING HOMEBRIDGE!
DS_HOSTNAME=DiskStation
```

CU,

Thorsten

**Comment by marcoraddatz** [comment]
That's what I wanted to suggest. Good luck!

**Comment by drtp** [comment]
Thank you Marco.

**Comment by drtp** [comment]
Unfortunately, neither modifying the .env file nor turning off Bonjour did help. I am still not able to find the Homebridge with the Home App, if it runs in a Docker container. Running the Homebridge with exactly the same config.json and plugins on a Raspi 3 immediately gives me access to the device. [Here](https://github.com/oznu/docker-homebridge/issues/35), I read something about possible issues with DiskStations having two ethernet ports. My DS718+ has two ports. Could this cause the problems?

**Comment by deadbone** [comment]
Hi drtp
Did you check the checkbox in your homebridge's container to use the same network like the host ? (not very sure about my english lol)
I've an old 713+ with 2 ethernet ports, all's good for me

**Comment by yubiao** [comment]
@drtp for the Bonjour service problem, some one has previously commented:

"I need to shutdown the docker image, disable Bonjour in DSM, start the docker image. Homekit devices are working fine. Afterwards I can activate Bonjour again and it keeps working. But as soon as the Docker image is restarted, it stops working and I need to repeat the process."

Try that see if it works for you.

**Comment by drtp** [comment]
> Did you check the checkbox in your homebridge's container to use the same network like the host ?

Yes, I did. No success so far.

**Comment by drtp** [comment]
> @drtp for the Bonjour service problem, some one has previously commented:

I know and I tried. Unfortunately, that did not have any influence on the behaviour either. In addition, I always deleted the persist folder and changed the username within the config.json file.

**Comment by drtp** [comment]
Unfortunately, I didn't manage it. Nothing, I tried, did help. Hence, I went back to my Raspi 3 installation. If someone has an idea how to solve the problem, I would be happy to know.

CU,

Thorsten

**Comment by xking3** [comment]
Same problem here with iOS 11.2.5
It does not recognize the the Homebridge. Before the update it worked fine. Everthing was perfect. 
First i tought it was some miscofigruation or some other problem. I denistalled the docker container end did a fresh setup. 
Everthing seames to be ok. The protocol gives me only these errors:
*** WARNING *** The program 'nodejs' uses the Apple Bonjour compatibility layer of Avahi...
...
But on an other poste it was written that this is not to mention. 
Could you please help. I realy liked the option of using the docker to conntect to the homecenter. The solution is perfect!

**Comment by marcoraddatz** [comment]
I'm sorry to hear that, but unfortunately I cannot do anything about it: I tried to reproduce this bug without success. Had to re-install Homekit accessories yesterday, but all my devices got detected without any problems.

**Comment by drtp** [comment]
Meanwhile, I have tried so many things. For instance, I renamed the hostname of my DiskStation, I deactivated all my other Docker containers, I changed again the username und deleted several times the persist folder, I changed the IPv6 settings, and so on and so on. Unfortunately, nothing helped. My iPhone X is not able to find the Homebridge within the Docker container. Since it works flawlessly in combination with my Raspberry Pi 3 using the same settings, I changed back to it again. If anybody finds a way to cope with this problem, I would be happy to hear. In addition, I have started several similar threads [here](http://www.synology-forum.de/showthread.html?90876-Homebridge-kann-einfach-nicht-gefunden-werden) and [here](https://forum.smartapfel.de/forum/thread/979-homebridge-im-docker-auf-synololgy-diskstation-home-app-unter-ios-11-2-1-findet/?postID=18274#post18274).

By the way, I believe that this issue has nothing to do with iOS but with some specialities of the docker and DS settings. Otherwise, it also should not work with the Homebridge installed on a Raspi.

**Comment by Sweetgenius19** [comment]
Have you tried changing the PIN? You might want to change the pin by 1 in the config.json and start with a fresh container.

**Comment by xking3** [comment]
I had the same though some day's ago but i could not test it so far. I will test that and report the result ASAP.

**Comment by marcoraddatz** [comment]
> I changed the IPv6 settings

The container doesn't work well with IPv6. Is it disabled for you local lan?

**Comment by xking3** [comment]
I will check that.

**Comment by drtp** [comment]
> The container doesn't work well with IPv6. Is it disabled for you local lan?

I tried both disabling and enabling IPv6. Unfortunately, there has been no success in either case. 

Someone wrote that the issue might result from the fact that the DS718+ has two Ethernet interfaces. However, I only use one of them. What about the other guys having the same issues? Does your DiskStation has two LAN ports as well?

**Comment by marcoraddatz** [comment]
My 415+ also has two lan ports. I use them with a bond and had no problems with it.

**Comment by oechslein** [comment]
Maybe it helps others:
My iOS devices also could not find Homebridge (running in Docker on Synology).
After I stopped the second docker image that was running ioBroker, all of them could find Homebridge.

Digging a bit into it, a network scan while running both docker images showed that my Synology was not found under its normal name "Diskstation" but under the ioBroker docker image name. 

I was able to start ioBroker docker image afterwards without any problems to the Homebridge connection.
